docs/lang/articles/basic/field.md

```diff
@@ -3,103 +3,128 @@ sidebar_position: 4
+
+When accessing a 0-D field `x`, use `x[None] = 0`, *not* `x = 0`.
+
 :::
 
-### Example
-An example might help you understand scalar fields.
-Assume you have a gray-scale image. At each point in the image, there would be a pixel value. The width and height of the image are similar to the `shape` of the Taichi scalar field. The pixel value (0-D scalar) is like the element of the Taichi scalar field. We could use the following code to generate a gray-scale image with random pixel values:
+```python
+# For a 0D field, you are required to use the index None even though it has only one element
+f_0d[None] = 10.0
+f_1d[0] = 1
+f_2d[1, 2] = 255
+f_3d[3, 3, 3] = 2.0
+```
 
-``` python {5}
-import taichi as ti
+As above mentioned, you can use a 2D scalar field to represent a 2D grid of values. The following code snippet creates and displays a 640&times;480 image with randomly-generated gray scales:
 
+```python
+import taichi as ti
 ti.init(arch=ti.cpu)
 width, height = 640,480
+# Create a 640x480 scalar field, each of its elements representing a pixel value (f32)
 gray_scale_image = ti.field(dtype=ti.f32, shape=(width, height))
 
 @ti.kernel
 def fill_image():
+  	# Fill the image with random gray
     for i,j in gray_scale_image:
         gray_scale_image[i,j] = ti.random()
 
 fill_image()
-
+# Set a GUI of same size as the gray-scale image
 gui = ti.GUI('gray-scale image with random values', (width, height))
 while gui.running:
     gui.set_image(gray_scale_image)
     gui.show()
 ```
 
 :::tip
-In earlier versions of Taichi, you could not allocate new fields after executing the first kernel. Since Taichi v0.8.0, you can use a new class `FieldsBuilder` for dynamic field allocation and destruction. For more details, please see [Field (advanced)](/lang/articles/advanced/layout.md).
+
+With Taichi versions earlier than v0.8.0, you cannot allocate new fields after executing a kernel. As of v0.8.0, you can use the `FieldsBuilder` class to dynamically allocate or destruct fields. See the [Field (advanced)](/lang/articles/advanced/layout) for more information.
+
 :::
 
+### Meta data
+
+Meta data provides the basic information of a scalar field. You can retrieve the data type and shape of a scalar field via its `shape` and `dtype` property:
+
+```python
+f_1d.shape  # (128)
+f_3d.dtype  # f32
+```
+
 ## Vector fields
-We are all living in a gravitational field, which is a vector field. At each position in 3D space, there is a gravity force vector. The gravitational field could be represented by:
+
+As the name suggests, vector fields are the fields whose elements are vectors.
+
+- You can use a vector field to represent an RGB image. Then, each of its elements is an (r, g, b) triple.
+- You can use a vector field to represent a volumetric field. Then, each of its elements can be the velocity of the corresponding particle.
+
+### Declaration
+
 ```python
-gravitational_field = ti.Vector.field(n=3, dtype=ti.f32, shape=(x, y, z))
+import taichi as ti
```

Comment:
nit: I think these `import taichi as ti; ti.init()` are excessive, and we don't need them in every code snippet.